packages/core/src/commands/insertContentAt.ts

```diff
@@ -57,12 +70,21 @@ export const insertContentAt: RawCommands['insertContentAt'] = (position, value,
       ...options,
     }
 
-    const content = createNodeFromContent(value, editor.schema, {
-      parseOptions: {
-        preserveWhitespace: 'full',
-        ...options.parseOptions,
-      },
-    })
+    let content: Fragment | ProseMirrorNode
+
+    try {
+      content = createNodeFromContent(value, editor.schema, {
+        parseOptions: {
+          preserveWhitespace: 'full',
+          ...options.parseOptions,
+        },
+        errorOnInvalidContent: options.errorOnInvalidContent ?? editor.options.enableContentCheck,
+      })
+    } catch (e) {
+      // Only ever throws if enableContentCheck or errorOnInvalidContent is true
+      editor.emit('contentError', { editor, error: e as Error })
+      return false
```

Comment:
Unsure here, if they supply invalid content what is expected in this command? Just to return false or to also call the onContentError callback? Maybe that only happens if `enabledContentCheck = true`

Reply:
I'm unsure if your thumbs up means you like how it is or want to change it to what I sai.

Reply:
Mh good question. I think just returning false is enough as long as invalid content is not inserted.